core/app/models/spree/calculator/tiered_percent.rb

```diff
@@ -22,8 +23,16 @@ class Calculator::TieredPercent < Calculator
 
     def compute(object)
       order = object.is_a?(Order) ? object : object.order
-      _base, percent = preferred_tiers.sort.reverse.detect{ |b, _| order.item_total >= b }
-      (object.amount * (percent || preferred_base_percent) / 100).round(2)
+
+      _base, percent = preferred_tiers.sort.reverse.detect do |b, _|
+        order.item_total >= b
+      end
+
+      if preferred_currency.casecmp(order.currency).zero?
```

Comment:
I noticed that @jhawthorn had switched to casecmp from the `upcase` comparison so I followed suit: https://github.com/solidusio/solidus/commit/4d2b5ba513ce4269eef638f706e4a293e47b8fc7